src/message/handlers/download.ts

```diff
@@ -19,6 +23,52 @@ const cancelDownloadTask = () => {
   }
 }
 
+/**
+ * Download a file via RNFetchBlob
+ */
+const download = async ({
+  fileUrl,
+  fileName,
+  directory,
+  getFetchConfig,
+  onFetchComplete
+}: {
+  fileUrl: string
+  fileName: string
+  directory: string
+  getFetchConfig: (filePath: string) => RNFetchBlobConfig
+  onFetchComplete?: (response: FetchBlobResponse) => Promise<void>
+}) => {
+  const filePath = directory + '/' + fileName
+
+  try {
+    fetchTask = RNFetchBlob.config(getFetchConfig(filePath)).fetch(
+      'GET',
+      fileUrl
+    )
+
+    // Do this while download is occuring
+    // TODO: The RNFetchBlob library is currently broken for download progress events on Android.
+    fetchTask.progress({ interval: 250 }, (received, total) => {
+      dispatch(setDownloadedPercentage((received / total) * 100))
+    })
+
+    const fetchRes = await fetchTask
+
+    // Do this after download is done
+    dispatch(setVisibility({ drawer: 'DownloadTrackProgress', visible: false }))
+
+    await onFetchComplete?.(fetchRes)
```

Comment:
This is sick! I didn't know you could use nullable functions like that. Just want to confirm that this actually does work on iOS (i.e, passes the parameter correctly?)

Reply:
The optional chaining operator definitely looks a little wonky on functions lol, but it's super useful!

And yup I just double checked that the param gets passed properly on ios and sharing + flushing is successful